src/seedu/addressbook/data/person/Person.java

```diff
@@ -25,6 +27,7 @@ public Person(Name name, Phone phone, Email email, Address address, UniqueTagLis
         this.email = email;
         this.address = address;
         this.tags = new UniqueTagList(tags); // protect internal tags from changes in the arg list
+        this.sequenceNumber = ++Person.nextSequenceNumber;
```

Comment:
Since you increment before assignment, the increment should be done to the current sequence number.  But since you name it `nextSequenceNumber`, the expected instruction sequence should be assign first, then increment.